geo/src/algorithm/affine_ops.rs

```diff
@@ -453,6 +453,34 @@ impl<U: CoordFloat> AffineTransform<U> {
     }
 }
 
+impl Index<&str> for AffineTransform {
+    type Output = f64;
+
+    /// Get elements from the AffineTransform matrix using string-based indexing.
+    ///
+    /// ```
+    /// let transform = AffineTransform::new(10.0, 0.0, 400_000.0, 0.0, -10.0, 500_000.0);
+    ///
+    /// let a: f64 = transform["a"];
+    /// let b: f64 = transform["b"];
+    /// let c: f64 = transform["xoff"];
+    /// let d: f64 = transform["d"];
+    /// let e: f64 = transform["e"];
+    /// let f: f64 = transform["yoff"];
+    /// ```
+    fn index(&self, idx: &str) -> &Self::Output {
+        match idx {
+            "a" => &self.0[0][0],
+            "b" => &self.0[0][1],
+            "xoff" => &self.0[0][2],
+            "d" => &self.0[1][0],
+            "e" => &self.0[1][1],
+            "yoff" => &self.0[1][2],
```

Comment:
Need to decide if using these names are ok, or maybe just do `a`, `b`, `c`, `d`, `e`, `f`. See discussion at #1158.